### PR TITLE
Content traversing workflows - Synthetic WF and Replication w Options

### PR DESCRIPTION
**Pull request by davidjgonzalez**
This PR contains mechanics to support Content Traversing workflows. This is useful in AEM 6.2 as WF Folders can now be selected for applying Workflow to.

* Content Visitor - Walks Page or Asset trees looking for cq:Pages and dam:Assets to process.

* SyntheticWorkflowWrapper - WF Process that executes a param WF Model as Synthetic WF. Allows for easy execution of Synth WF via normal WF UIs though does require invocation of a WF. Can be used to traverse content trees.

* Replication w Options - performs a Content Traversing replication w replication options passed in via PROCESS_ARGS

### CODE REVIEW

File: bundle/src/main/java/com/adobe/acs/commons/workflow/process/impl/ReplicateWithOptionsWorkflowProcess.java [hunk 1]
```diff
@@ -0,0 +1,164 @@
...
+    public void execute(WorkItem workItem, WorkflowSession workflowSession, MetaDataMap metaDataMap) throws WorkflowException {
+        ResourceResolver resourceResolver = null;
+
+        try {
+            resourceResolver = workflowHelper.getResourceResolver(workflowSession);
+            final String originalPayload = (String) workItem.getWorkflowData().getPayload();
+            final List<String> payloads = workflowPackageManager.getPaths(resourceResolver, originalPayload);
+            final ProcessArgs processArgs = new ProcessArgs(metaDataMap);
+
+            final AtomicInteger count = new AtomicInteger(0);
```
Comment (bstopp) [inline bundle/src/main/java/com/adobe/acs/commons/workflow/process/impl/ReplicateWithOptionsWorkflowProcess.java]: This is created/instantiated, but never consumed.

**Comment by bstopp** [comment]
The _SyntheticWrapperWorkflowProcess_ and _ReplicateWithOptionsWorkflowProcess_ have a lot of shared/duplicated logic; is there no way to remove some of the C/P/D risk?

**Comment by davidjgonzalez** [comment]
@bstopp agree on the code dupe; started pulling it out but realized 1) it really seems useful to only these 2 use cases  2) its the same but every other line is slightly different and i'd have to make a bunch of partial abstractions, which creates a bunch of supporting files for 2 otherwise simple WFs

File: bundle/src/main/java/com/adobe/acs/commons/workflow/process/impl/ReplicateWithOptionsWorkflowProcess.java (hunk 1 above)
Comment (davidjgonzalez) [inline bundle/src/main/java/com/adobe/acs/commons/workflow/process/impl/ReplicateWithOptionsWorkflowProcess.java]: Fixed

**Pull request merged**